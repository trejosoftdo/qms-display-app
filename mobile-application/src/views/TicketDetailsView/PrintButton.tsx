import * as React from 'react';
import { useTranslation } from 'react-i18next';
import ActionButton from '../../common/components/ActionButton';
import { TRANSLATION_PRINT_TICKET_KEY } from '../../common/translations/translation-keys';

/**
 * PrintButtonProps defines the props for the Printing Form Component.
 *
 * @property onPress - on press handler
 */
interface PrintButtonProps {
  onPress: () => void;
}

/**
 * A component that represents a button for printing ticket details
 *
 * @param {PrintButtonProps} props - The props for the Print Button component.
 */
const PrintButton: React.FC<PrintButtonProps> = (props: PrintButtonProps) => {
  const { t } = useTranslation();
  return (
    <ActionButton
      icon="printer"
      onPress={props.onPress}
      message={t(TRANSLATION_PRINT_TICKET_KEY)}
    />
  );
};

export default PrintButton;
