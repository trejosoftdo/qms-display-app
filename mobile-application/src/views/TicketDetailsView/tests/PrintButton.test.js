import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import PrintButton from '../PrintButton';


describe('PrintButton component', () => {
  const mockProps = {
    onPress: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  })

  it('renders the component as expected with the given props', () => {
    expect(renderShallow(<PrintButton {...mockProps} />)).toMatchSnapshot();
  });

  it('calls the onPress handler when being clicked', () => {
    const { getByRole } = render(<PrintButton {...mockProps} />);
    const element = getByRole('button');
    fireEvent.press(element);
    expect(mockProps.onPress).toHaveBeenCalled();
  });
});
