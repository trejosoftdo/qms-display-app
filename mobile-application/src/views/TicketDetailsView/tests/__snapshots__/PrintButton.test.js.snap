// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintButton component renders the component as expected with the given props 1`] = `
<ActionButton
  icon="printer"
  message="Translated[translation:printTicket]({})"
  onPress={[MockFunction]}
/>
`;
